cohortextractor/validate_dummy_data.py

```diff
@@ -0,0 +1,131 @@
+from datetime import datetime
+
+import pandas as pd
+
+
+SUPPORTED_FILE_FORMATS = ["csv", "csv.gz"]
+
+
+class DummyDataValidationError(Exception):
+    pass
+
+
+def validate_dummy_data(column_definitions, dummy_data_file):
+    """Validate that dummy data provided by user matches expected structure and format.
+
+    Raises DummyDataValidationError if dummy data is not valid.
+    """
+
+    df = read_into_dataframe(dummy_data_file)
+    validate_expected_columns(df, column_definitions)
+
+    for col_name, query_node in column_definitions.items():
+        if col_name == "population":
+            # Ignore the population definition, since it is not used as a column in the
+            # output
```

Comment:
We do this ignoring in two places. Could we just strip it out of the column definitions before validating?